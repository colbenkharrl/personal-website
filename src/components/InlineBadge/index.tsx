import { ComponentPropsWithoutRef, FC, ReactNode } from 'react';
import clsx from 'clsx';

export type InlineBadgeProps = ComponentPropsWithoutRef<'span'> & {
  children: ReactNode;
};

export const InlineBadge: FC<InlineBadgeProps> = ({
  children,
  className,
  ...props
}) => {
  return (
    <span
      className={clsx(
        className,
        'inline-flex items-center rounded-md bg-amber-50 px-2 py-1 text-xs font-medium text-amber-600 ring-1 ring-inset ring-amber-500/10 hover:bg-amber-100 dark:bg-amber-400/10 dark:text-amber-400 dark:ring-amber-400/30 dark:hover:bg-amber-400/20',
      )}
      {...props}
    >
      {children}
    </span>
  );
};
