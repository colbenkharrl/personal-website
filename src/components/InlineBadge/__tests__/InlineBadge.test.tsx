import { render, screen } from '@testing-library/react';
import * as stories from '../__stories__/InlineBadge.stories';
import { composeStories } from '@storybook/react';

const { Basic } = composeStories(stories);

it('InlineBadge matches snapshot (regression test)', () => {
  render(<Basic data-testid="InlineBadge" />);
  expect(screen.getByTestId('InlineBadge')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
