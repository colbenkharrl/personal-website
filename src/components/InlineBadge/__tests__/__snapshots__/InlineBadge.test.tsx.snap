// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineBadge matches snapshot (regression test): Full component snapshot. 1`] = `
<span
  class="inline-flex items-center rounded-md bg-amber-50 px-2 py-1 text-xs font-medium text-amber-600 ring-1 ring-inset ring-amber-500/10 hover:bg-amber-100 dark:bg-amber-400/10 dark:text-amber-400 dark:ring-amber-400/30 dark:hover:bg-amber-400/20"
  data-testid="InlineBadge"
>
  Text.
</span>
`;
